### PR TITLE
[bitnami/tomcat] Add existing secret management

### PR DESCRIPTION
**Pull request by matthrobin**
### Description of the change

Add existing secret management

### Benefits

Be able to use an existing kubernetes secret to set the tomcat password

### Possible drawbacks

No drawback identified

### Applicable issues

No applicable issue

### Additional information

Tested locally to prevent any regression

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/tomcat/values.yaml [hunk 1]
```diff
@@ -92,6 +92,9 @@ tomcatUsername: user
 ## ref: https://github.com/bitnami/containers/tree/main/bitnami/tomcat#creating-a-custom-user
 ##
 tomcatPassword: ""
+## @param existingSecret Use existing secret for password details (`tomcatPassword` will be ignored and picked up from this secret). The secret has to contain the keys `tomcat-password`
```
Comment (juan131) [inline bitnami/tomcat/values.yaml:95]: Small typo:
```suggestion
## @param existingSecret Use existing secret for password details (`tomcatPassword` will be ignored and picked up from this secret). The secret has to contain the key `tomcat-password`
```

File: bitnami/tomcat/templates/secrets.yaml [hunk 2]
```diff
@@ -15,3 +16,4 @@ metadata:
 type: Opaque
 data:
   tomcat-password: {{ include "common.secrets.passwords.manage" (dict "secret" (include "common.names.fullname" .) "key" "tomcat-password" "providedValues" (list "tomcatPassword") "length" 10 "strong" false "context" $) }}
+{{- end }}
```
Comment (juan131) [inline bitnami/tomcat/templates/secrets.yaml:19]: Please ensure you include a "new line" character at the end of file as suggested by GitHub

**Review by juan131 — COMMENTED** [review]
Thanks so much for this PR, we really appreciate it! Please take a look to my suggestions (very minor things) when you have a chance.

File: bitnami/tomcat/templates/secrets.yaml (hunk 2 above)
Comment (matthrobin) [reply bitnami/tomcat/templates/secrets.yaml:19]: Done ! thanks for the feedback

File: bitnami/tomcat/values.yaml (hunk 1 above)
Comment (matthrobin) [reply bitnami/tomcat/values.yaml]: Done ! Thanks for the feedback

**Pull request merged**